plugins/trace_api_plugin/include/eosio/trace_api/request_handler.hpp

```diff
@@ -42,12 +43,15 @@ namespace eosio::trace_api {
          }
 
          yield();
-
-         auto data_handler = [this](const action_trace_v0& action, const yield_function& yield) -> fc::variant {
-            return data_handler_provider.process_data(action, yield);
+         
+         auto data_handler_v0 = [this](const action_trace_v0& action, const yield_function& yield) -> std::tuple<fc::variant, fc::optional<fc::variant>> {
```

Comment:
Then this could become,
``` c++
auto data_handler = [this](const auto& action, const yield_function& yield) -> std::tuple<fc::variant, std::optional<fc::variant>> {
   return std::visit([](const auto& trace) {
       return data_handler_provider.process_data(trace, yield);
       }, action);
}
```

Or something like that.

Reply:
will try.